src/fshtypes/ValueSet.ts

```diff
@@ -0,0 +1,22 @@
+import { FshEntity } from './FshEntity';
+import { ValueSetTerm } from './ValueSetTerm';
+import { ValueSetFilter } from './ValueSetFilter';
+
+/**
+ * For more information about the composition of a ValueSet,
+ * @see {@link http://hl7.org/fhir/valueset-definitions.html#ValueSet.compose}
+ */
+export class ValueSet extends FshEntity {
+  id: string;
+  url?: string;
+  description?: string;
+  inclusions: (ValueSetTerm | ValueSetFilter | ValueSet)[];
+  exclusions: (ValueSetTerm | ValueSetFilter | ValueSet)[];
```

Comment:
I wonder if we should model this more closely to how FHIR defines it?  Would it be simpler that way?
```ts
// ...
inclusions: ValueSetRule[]
exclusions: ValueSetRule[]
// ...

class ValueSetRule extends FshEntiry {
  public system?: string;
  public rule?: ValueSetConcept | ValueSetFilter | ValueSetReference;
  // with...
}

type ValueSetReference: string;

Reply:
Then you would not need to put the `system` into the other classes.

There is a question of whether `rule` should be an array or not -- because technically, in the FHIR spec, you can put all the rules that share a code system into the same `include` item.  But probably, to be consistent w/ the FSH grammar, it makes more sense as a rule per line.